libkbfs/folder_branch_ops.go

```diff
@@ -7901,6 +8224,57 @@ func (fbo *folderBranchOps) SetSyncConfig(
 		newConfig.Paths = paths
 	}
 
+	oldConfig, err := fbo.getProtocolSyncConfig(ctx, lState, md)
+	if err != nil {
+		return nil, err
+	}
+
+	oldPartial := oldConfig.Mode == keybase1.FolderSyncMode_PARTIAL
+	newPartial := newConfig.Mode == keybase1.FolderSyncMode_PARTIAL
+	if oldPartial && !newPartial {
+		if fbo.markAndSweepTrigger == nil {
+			return nil, errors.New(
+				"Unexpected sync config; mark-and-sweep already started")
+		}
+
+		fbo.log.CDebugf(ctx, "Exiting partial mode, stopping mark-and-sweep")
+		close(fbo.markAndSweepTrigger)
+		fbo.markAndSweepTrigger = nil
```

Comment:
I'm confused. Shouldn't we be marking and sweeping out all old blocks if `newPartial` is false?

Reply:
If the new config is enabling syncing, then a full sync will be kicked off, subsuming the currently synced blocks. 
 If the new config is disabling syncing completely, then `ConfigLocal.SetTlfSyncState()` calls `ClearAllTlfBlocks()` and gets rid of everything, without needing mark-and-sweep.